radixdlt-core/radixdlt/src/main/java/com/radixdlt/consensus/epoch/EpochManager.java

```diff
@@ -342,26 +328,22 @@ private void processConsensusEventInternal(ConsensusEvent consensusEvent) {
   }
 
   public void processConsensusEvent(ConsensusEvent consensusEvent) {
-    if (consensusEvent.getEpoch() > this.currentEpoch()) {
-      log.debug(
-          "{}: CONSENSUS_EVENT: Received higher epoch event: {} current epoch: {}",
-          this.self::getSimpleName,
-          () -> consensusEvent,
-          this::currentEpoch);
+    if (consensusEvent.getEpoch() != this.currentEpoch()) {
 
-      // queue higher epoch events for later processing
-      // TODO: need to clear this by some rule (e.g. timeout or max size) or else memory leak attack
-      // possible
-      queuedEvents
-          .computeIfAbsent(consensusEvent.getEpoch(), e -> new ArrayList<>())
-          .add(consensusEvent);
-      counters.increment(CounterType.EPOCH_MANAGER_QUEUED_CONSENSUS_EVENTS);
-      return;
-    }
+      if (consensusEvent.getEpoch() == this.currentEpoch() + 1) {
+
+        // Keep only events for the last view
+        if (isNotTheSameViewAs(consensusEvent)) {
```

Comment:
I'm not sure I understand what this is doing. It allow allows view events to be added for the same view as the first in the list. What if a view 2 event came in before the view 1 event?

Could we instead (eg) restrict ourselves to only tracking events for (eg) the first 20 rounds of the next epoch? (enough for a couple of timeouts). We could even store these in a (round -> event) map if we wished to

Reply:
Actually, I've found that this version does not pass some integration tests. I've removed check and let the cache hold all events which belong to next epoch.